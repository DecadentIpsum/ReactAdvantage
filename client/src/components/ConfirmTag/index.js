import React from 'react';
import classnames from 'classnames';

import './index.css';

const ConfirmTag = props => {
    const { active, ...restProps } = props;
    const classNames = classnames({
        'confirm-tag': true,
        inactive: !active,
    });

    return (
        <div className={classNames} {...restProps}>
            {props.children}
        </div>
    );
}

export default ConfirmTag;