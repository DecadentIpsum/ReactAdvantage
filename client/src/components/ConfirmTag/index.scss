@import "global-styles/common.scss";

.confirm-tag {
    display: inline-block;
    width: 30px;
    padding: 0.25em 0;
    background: $success;
    color: $white;
    border-radius: 3px;
    text-align: center;

    &.inactive {
        background: $error;
    }
}