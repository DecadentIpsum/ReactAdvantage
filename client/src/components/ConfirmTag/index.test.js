import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import ConfirmTag from './index';

describe('it renders correctly', () => {
    test('active ConfirmTag renders correctly', () => {
        const component = shallow(<ConfirmTag />);

        expect(component).toMatchSnapshot();
    });

    test('inactive ConfirmTag renders correctly', () => {
        const component = shallow(<ConfirmTag active={false} />);

        expect(component).toMatchSnapshot();
    });
});
