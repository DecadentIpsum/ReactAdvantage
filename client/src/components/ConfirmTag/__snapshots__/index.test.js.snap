// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly active ConfirmTag renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmTag />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "confirm-tag inactive",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "confirm-tag inactive",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`it renders correctly inactive ConfirmTag renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmTag
    active={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "confirm-tag inactive",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "confirm-tag inactive",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
